### PR TITLE
Drop batch/v2alpha1

### PR DESCRIPTION
**Pull request by soltysh**
**What type of PR is this?**

/kind cleanup
/kind api-change
/kind deprecation

**What this PR does / why we need it**:
This is part of the work related with promoting CronJobs to GA. See https://github.com/kubernetes/enhancements/issues/19

**Special notes for your reviewer**:
/assign @liggitt @deads2k 

**Does this PR introduce a user-facing change?**:
```release-note
The `batch/v2alpha1` CronJob type definitions and clients are deprecated and removed.
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:
```docs
- [KEP]: https://github.com/kubernetes/enhancements/tree/master/keps/sig-apps/19-Graduate-CronJob-to-Stable 
```

### CODE REVIEW

**Comment by soltysh** [comment]
/hold
I think there are still some leftovers in the infra. I'll look for them.

**Comment by spiffxp** [comment]
The perception that updating release-branch job configs (https://github.com/kubernetes/test-infra/pull/20879) is what allowed this to merge felt _really wrong_ to me.  Did some digging.

What _actually_ allowed these to pass wasn't just removing the explicit `--runtime-config=batch/v2alpha1=true` flags from job configs, but also the (poorly chosen) default of `batch/v2alpha1=true` hardcoded into `kubetest` (ref: https://github.com/kubernetes/test-infra/pull/20865/files#diff-71dd7bf85cfecc6c2c2fa0956ad5f182433bcc0f287aaa83f2052eb3f8ba9710L175)

Changes to `kubetest` don't take effect until a new `kubekins` image is built and job configs are updated to use the new image.

https://github.com/kubernetes/test-infra/pull/20852 is the PR that bumped job configs to the kubekins image including the `kubetest` change.  It merged about 20 hours ago.  This lines up with when the jobs started passing.

tl;dr release-branch job changes had nothing to do with it, they just happened to land near the time the `kubestest` changes were actually rolled out

**Comment by spiffxp** [comment]
>Changes to kubetest don't take effect until a new kubekins image is built and job configs are updated to use the new image

This needs to be made clearer.  We roll out changes this way to allow someone to canary kubetest changes before they roll out to all jobs. @liggitt suggested having kubetest print out its version

**Comment by BenTheElder** [comment]
+1, but besides printing version I recommend recording it to an artifact. Kind does this when log dumping and it's been super useful.

**Comment by spiffxp** [comment]
Opened https://github.com/kubernetes/test-infra/issues/20886
